### ISSUE TITLE
Build error with gcc 10.2?

### ISSUE THREAD

**Issue by sefeng211** [mozilla/rr#2660]
I got some build errors for some test files when I build `rr` against the latest master (b8dba6b718deca87f38ab487266eb973ddf381da)

I am running `gcc 10.2.0-2`

```
/home/sefeng/.local/workspace/repos/.other/rr/src/test/nsutils.h:46:24: error: argument 2 is null but the corresponding size argument 3 value is 1 [-Werror=nonnull]
   46 |       test_assert(0 == read(child_block[0], NULL, 1));
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sefeng/.local/workspace/repos/.other/rr/src/test/nsutils.h:46:7: note: in expansion of macro ‘test_assert’
   46 |       test_assert(0 == read(child_block[0], NULL, 1));
      |       ^~~~~~~~~~~
```

Thanks!

**Comment by Keno** [comment]
GCC is of course technically correct, but here we're explicitly expecting the pipe to be closed before any bytes are written. To silence the warning, we could either use the read syscall directly or just give it a dummy buffer to make GCC happy.

**Comment by sefeng211** [comment]
Thanks for fixing the previous. There are two more 
```
/home/sefeng/.local/workspace/repos/.other/rr/obj/32/read_oversize.c: In function ‘main’:
/home/sefeng/.local/workspace/repos/.other/rr/obj/32/read_oversize.c:15:9: error: ‘read’ specified size 4294963200 exceeds maximum object size 2147483647 [-Werror=stringop-overflow=]
   15 |   ret = read(fd, buf, UINTPTR_MAX - 0xfff);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/sefeng/.local/workspace/repos/.other/rr/obj/32/util.h:103,
                 from /home/sefeng/.local/workspace/repos/.other/rr/obj/32/read_oversize.c:3:
/usr/include/unistd.h:360:16: note: in a call to function ‘read’ declared with attribute ‘write_only (2, 3)’
  360 | extern ssize_t read (int __fd, void *__buf, size_t __nbytes) __wur
      |                ^~~~
/home/sefeng/.local/workspace/repos/.other/rr/src/test/read_oversize.c: In function ‘main’:
/home/sefeng/.local/workspace/repos/.other/rr/src/test/read_oversize.c:15:9: error: ‘read’ specified size 18446744073709547520 exceeds maximum object size 9223372036854775807 [-Werror=stringop-overflow=]
   15 |   ret = read(fd, buf, UINTPTR_MAX - 0xfff);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/sefeng/.local/workspace/repos/.other/rr/src/test/util.h:103,
                 from /home/sefeng/.local/workspace/repos/.other/rr/src/test/read_oversize.c:3:
/usr/include/unistd.h:360:16: note: in a call to function ‘read’ declared with attribute ‘write_only (2, 3)’
  360 | extern ssize_t read (int __fd, void *__buf, size_t __nbytes) __wur
      |                ^~~~

```

**Comment by rocallahan** [comment]
Can you disable -Werror to collect them all?

**Comment by sefeng211** [comment]
Sorry about that, I should've done that in the first place. This is the full build log https://pastebin.com/raw/ifrsBCBJ

Those two should be the only ones left. 
```
➜  obj grep -i "warning" build.log
/home/sefeng/.local/workspace/repos/.other/rr/obj/32/read_oversize.c:15:9: warning: ‘read’ specified size 4294963200 exceeds maximum object size 2147483647 [-Wstringop-overflow=]
/home/sefeng/.local/workspace/repos/.other/rr/src/test/read_oversize.c:15:9: warning: ‘read’ specified size 18446744073709547520 exceeds maximum object size 9223372036854775807 [-Wstringop-overflow=]
/usr/bin/ld: CMakeFiles/fxregs_32.dir/32/x86/fxregs.c.o: warning: relocation in read-only section `.text'
/usr/bin/ld: warning: creating DT_TEXTREL in a PIE
```
(Not sure what that relocation warning is)

**Comment by rocallahan** [comment]
fixed in 2e8d01014e57f5fa8cf0de0c7a47ee9697da5012
